### PR TITLE
roccat-tools: init at 3.7.1

### PR DESCRIPTION
**Pull request by heydojo**
- Created a new expression for roccat-tools
- The software provides access to features found
  on roccat.org peripheral devices; such as mice
  and keyboards.
- The software was tested using a ROCCAT® KONE PURE MILITARY
  wired usb mouse :
  http://roccat.org/en-GB/Products/Gaming-Mice/Kone-Pure-Series/Kone-Pure-Military/
  and successfully changes the device's settings.
- roccat-tools allows the user to configure the following ROCCAT®
  products :

ROCCAT® Arvo
ROCCAT® Kone
ROCCAT® Kone Pure Military
ROCCAT® Kone XTD
ROCCAT® Kone XTD Optical
ROCCAT® Nyth
ROCCAT® Ryos TKL
ROCCAT® Ryos MK
ROCCAT® Isku
ROCCAT® Kone Plus
ROCCAT® Kone Pure
ROCCAT® Kone Pure Optical
ROCCAT® Kova Plus
ROCCAT® Pyra
ROCCAT® Savu
ROCCAT® Isku FX
ROCCAT® Lua
ROCCAT® Tyon

The project http://sourceforge.net/projects/roccat/
is not supported by ROCCAT® but roccat did provide
the developer with peripherals in order to encourage
and ease his development of the software.

A roccat module also exists in the Linux® kernel
and was developed by the author of roccat-tools.

As a side note I will mention that the ROCCAT® device which
was tested is a very high quality product and I encourage
anyone who is in the market for a new mouse to consider
one due their positive attitude towards Linux® users
and sponsorship of e-sports.

### CODE REVIEW

File: pkgs/os-specific/linux/roccat-tools/default.nix [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+        genericBuild
+
+      )
+    '';
+
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    description = "Roccat hardware support for Linux";
```
Comment (jagajaga) [inline pkgs/os-specific/linux/roccat-tools/default.nix]: Please omit pkg's name in the description.

**Comment by jagajaga** [comment]
Please rename commit to `roccat-tools: init at <version>`

File: pkgs/os-specific/linux/roccat-tools/default.nix (hunk 1 above)
Comment (jgillich) [inline pkgs/os-specific/linux/roccat-tools/default.nix]: @jagajaga Uhm, if we remove "Roccat" from the description, it won't make much sense anymore. But maybe something like `Various tools for Roccat hardware` is more descriptive..

Comment (jagajaga) [inline pkgs/os-specific/linux/roccat-tools/default.nix]: You are right.

File: pkgs/os-specific/linux/roccat-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	    substituteInPlace $out/share/applications/roccatpyraconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatsavuconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatiskufxconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatkonepureconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatkonextdconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatluaconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatryosmkconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccattyonconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    touch --no-create /usr/share/icons/hicolor
+	    # gtk-update-icon-cache -q /usr/share/icons/hicolor
```
Comment (jagajaga) [inline pkgs/os-specific/linux/roccat-tools/default.nix]: Do we need these commented lines?

File: pkgs/os-specific/linux/roccat-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	    substituteInPlace $out/share/applications/roccatkonepureopticalconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatkovaplusconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatpyraconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatsavuconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatiskufxconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatkonepureconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatkonextdconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatluaconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccatryosmkconfig.desktop --replace "Terminal=false" "Terminal=true"
+	    substituteInPlace $out/share/applications/roccattyonconfig.desktop --replace "Terminal=false" "Terminal=true"
```
Comment (jgillich) [inline pkgs/os-specific/linux/roccat-tools/default.nix]: I think these should really be a loop over all .desktop files.

Comment (heydojo) [inline pkgs/os-specific/linux/roccat-tools/default.nix]: Yeah.

**Comment by heydojo** [comment]
@jagajaga @jgillich could you please let me know if
```
maintainers = [ maintainers.heydojo ];
```
is correct?

@nckx is using a different stanza in a recent commit d1440745c6e6f6ee611679dc30b40cb963724427 and it looks like maintainers.heydojo is blocking a jenkins pass.
Thanks

**Comment by nckx** [comment]
@heydojo: no, your syntax is fine and entirely equivalent to `with maintainers`. I use the latter for ideological reasons (‘maintainers are good; have many’).

Did you add yourself to `lib/maintainers.nix`?

File: pkgs/os-specific/linux/roccat-tools/default.nix [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  toolsSrc = fetchurl {
+    url = "mirror://sourceforge/roccat/${name}.tar.bz2";
+    sha256 = "b816eab240e8afc8f3fa5509ca523aab04994606cd6d6beb8dbf8a65dd25daca";
+  };
+
+  dontUseCmakeBuildDir=true;
+
+  nativeBuildInputs = [ cmake pkgconfig ];
+
+  buildInputs = [ stdenv cmake dbus glib libcanberra atk cairo gdk_pixbuf gtk2 pango libnotify libgudev xorg.libX11 xorg.libpthreadstubs xorg.libXdmcp xorg.libxshmfence sqlite gegl dbus_glib gettext pkgconfig ];
```
Comment (nckx) [inline pkgs/os-specific/linux/roccat-tools/default.nix]: Please wrap long lines like these whenever possible (monolithic strings like `config2` below are tolerable).

File: pkgs/os-specific/linux/roccat-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  builder = builtins.toFile "builder.sh"
+    ''
+	export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:$out/usr/share/pkgconfig"
+      # First build libgaminggear.
+      (
+        source $stdenv/setup
+        cmakeFlags=$config1
+	postInstall () {
+	      mkdir -p "$out/etc/udev/rules.d/"
```
Comment (nckx) [inline pkgs/os-specific/linux/roccat-tools/default.nix]: ^ Indentation is wonky.

File: pkgs/os-specific/linux/roccat-tools/default.nix [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    '';
+
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    description = "Tools for ROCCAT® hardware";
+    homepage = http://sourceforge.net/projects/roccat/;
+    license = licenses.gpl2Plus;
+    maintainers = [ maintainers.heydojo ];
+    platforms = stdenv.lib.platforms.linux;
```
Comment (nckx) [inline pkgs/os-specific/linux/roccat-tools/default.nix]: You already brought `stdenv.lib` into scope, so it isn't needed here.

**Comment by nckx** [comment]
Oh, and go ahead and squash all follow-up commits into one.

**Comment by heydojo** [comment]
I will re-evaluate my proposed merge and re-submit when I have made further suitable adjustments.
ありがとうございます

**Comment by nckx** [comment]
@heydojo: what was wrong with this PR?

**Comment by heydojo** [comment]
I unfortunately don't have enough time to get this merged right now.
I started a new job :)

**Comment by nckx** [comment]
No problem. Congratulations! :cake:

**Comment by heydojo** [comment]
:+1: Cheers!
It's at a really great local IT company and I was offered the job on my Birthday. Double celebration!

I will try to come back with this merge once I settle in.

**Pull request closed without merging**